### DISCUSSION TITLE
Missing some search results

### DISCUSSION THREAD

**Discussion by JessMeats** [microsoft-search/pnp-modern-search#1751] [Q&A]
I've configured the search results web part to act as a "related content" view on news items. It's supposed to show pages based on the metadata properties of the current page. 

This is what I want as the query: 
PromotedStateOWSNMBR:2 AND RefinableString08:{Page.News_x0020_Department} AND RefinableString09:{Page.News_x0020_Category} NOT path:{Page.URL}

This should give me published news items that have the same department and category tags as the current news item, but with the actual current page excluded so that it doesn't recommend itself. But in some cases, this isn't finding any results even though there are items that match the query. 

I've tried amending the query to only include one or other of the refinable string properties and it will then give me some results - but not the full list it should. In one of my tests, including just the department filter found 5 results, and searching by category found 2 results - without any overlap between the two result sets. However one of the results found in the department search also had the same category, and one of the results in the category search also had the same department. They just weren't showing up in the other search results. And looking in the pages library for published news items with these properties showed other items that weren't showing up as results. 

So if we have a hypothetical page with properties category = CatA and department = Dep1. 
Putting this query: 
PromotedStateOWSNMBR:2 AND RefinableString09:{Page.News_x0020_Category} NOT path:{Page.URL}
Would find two results. Result 1 is CatA and Dep2. Result 2 is CatA and Dep1. 

Putting this query: 
PromotedStateOWSNMBR:2 AND RefinableString08:{Page.News_x0020_Department} NOT path:{Page.URL}
Would result in 5 results, none of which is result 1 or 2, even though result 2 ought to be a match. And Result 3 is CatA and Dep1 and so should have shown up for the other query but didn't. 

Going into the pages library and filtering by these properties directly in the library would show other results for either of the partial queries that didn't show up either. 

I'm not sure why some of the search results are just not being found.

**Comment by wobba** [comment]
You can use `PromotedState:2` to get published news pages. And without knowing if your items has the values in `RefinableStringXX` it's hard to troubleshoot. And you don't mention if you mapped cp's to mp's at a site level or at tenant level.

**Reply by JessMeats** [reply]
The results are all published news - so I don't think that's the problem. The two metadata columns in question have been mapped to crawled and managed properties at the tenant level.  

The department column is mapped to the RefinableString08 field and the category column is mapped to RefinableString09. Both columns are Choice type and mandatory. There are currently around 140 news items in the pages library, all of which have a value set in each of these columns. 

The whole point of the problem, as described above, was that there were news pages that matched the properties - i.e. they had the values in those columns that were the same as the current page - that didn't show up in the search results when they should have done. The web part is finding some of the results that matched the query, but not all of them, and it is sometimes showing 0 results when there should have been at least 2 or 3 if I go and manually look for pages where the properties match.

**Reply by wobba** [reply]
You have to prove the items have values, and if not that's a support case with Microsoft. If items don't have the values in the index, then search will not work to query for them on those properties.

Also, ensure you haven't configured yourself in this scenario https://www.techmikael.com/2020/03/side-effect-of-mapping-crawled.html
